[TIMOB-24415] Android: Fix a11y properties for ListItems

drauggres opened the pull request:
**JIRA:** https://jira.appcelerator.org/browse/TIMOB-24415

This PR uses same [base commit](https://github.com/appcelerator/titanium_mobile/commit/25628c1a39f27f3901cb71a70de6bec2b349f2ec) as https://github.com/appcelerator/titanium_mobile/pull/10397

build commented:
<!--
  1 failure:  :microscope: Ther...
  0 warning: 
  1 messages
  
  DangerID: danger-id-default;
-->

<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true">Fails</th>
    </tr>
  </thead>
  <tbody><tr>
      <td>:no_entry_sign:</td>
      <td>

  :microscope: There are library changes, but no changes to the unit tests. That's OK as long as you're refactoring existing code, but will require an admin to merge this PR. Please see <a href="https://github.com/drauggres/titanium_mobile/blob/TIMOB-24415/README.md#unit-tests">README.md#unit-tests</a> for docs on unit testing.
  </td>
    </tr>
  </tbody>
</table>



<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true">Messages</th>
    </tr>
  </thead>
  <tbody><tr>
      <td>:book:</td>
      <td>

  :tada: Another contribution from our awesome community member, drauggres! Thanks again for helping us make Titanium SDK better. :thumbsup:
  </td>
    </tr>
  </tbody>
</table>


<p align="right">
  Generated by :no_entry_sign: <a href="http://github.com/danger/danger-js/">dangerJS</a>
</p>

drauggres commented:
About "needs tests" label: we can't create unit tests to check this feature. For proper testing we need UI tests (and smth. like appium). Same for #10397.

build commented:
<!--
  2 failure:  :microscope: Ther..., Tests have failed...
  0 warning: 
  3 messages
  1 markdown notices
  DangerID: danger-id-Danger;
-->

<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true">Fails</th>
    </tr>
  </thead>
  <tbody><tr>
      <td>:no_entry_sign:</td>
      <td>

  :microscope: There are library changes, but no changes to the unit tests. That's OK as long as you're refactoring existing code, but will require an admin to merge this PR. Please see <a href="https://github.com/drauggres/titanium_mobile/blob/TIMOB-24415/README.md#unit-tests">README.md#unit-tests</a> for docs on unit testing.
  </td>
    </tr>
  
<tr>
      <td>:no_entry_sign:</td>
      <td>Tests have failed, see below for more information.</td>
    </tr>
  </tbody>
</table>



<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true">Messages</th>
    </tr>
  </thead>
  <tbody><tr>
      <td>:book:</td>
      <td>

  :floppy_disk: <a href="https://jenkins.appcelerator.org/job/titanium-sdk/job/titanium_mobile/job/PR-10399/3/artifact/dist/mobilesdk-8.1.0.v20190513043303-osx.zip">Here's the generated SDK zipfile</a>.
  </td>
    </tr>
  
<tr>
      <td>:book:</td>
      <td>:tada: Another contribution from our awesome community member, drauggres! Thanks again for helping us make Titanium SDK better. :thumbsup:</td>
    </tr>
  
<tr>
      <td>:book:</td>
      <td>:x: 1 tests have failed
There are 1 tests failing and 466 skipped out of 3727 total tests.</td>
    </tr>
  </tbody>
</table>

### Tests:

<table><tr><th>Classname</th><th>Name</th><th>Time</th><th>Error</th></tr>
<tr><td>android.emulator.Titanium.Media.AudioPlayer</td><td>#restart</td><td>27.883</td><td><pre>Error: timeout of 5000ms exceeded
at Titanium.&lt;anonymous&gt; (/ti-mocha.js:4326:18)</pre></td></tr>
</table>

<p align="right">
  Generated by :no_entry_sign: <a href="https://danger.systems/js">dangerJS</a> against 165ad5eeefac09fe73b3e11ddcea39b849e67a88
</p>

drauggres committed:
feat(android): add accessibility properties to MenuItem

Add "accessibilityHint", "accessibilityLabel" and "accessibilityValue" properties to MenuItem
element.
Make 'TiUIView.composeContentDescription' static.

refs TIMOB-26463

drauggres committed:
fix(android): accessibility properties for ListItem

- Remove hardcoded content description.
- Add a11y properties to the refresh list.
Need to refresh all a11y properties if any of them are changed, because native "contentDescription"
property is build as their concatenation.

refs TIMOB-24415

drauggres commented:
```javascript
var win = Ti.UI.createWindow({backgroundColor: 'white'});

var myTemplate = {
    childTemplates: [
        {                            // Image justified left
            type: 'Ti.UI.ImageView', // Use an image view for the image
            bindId: 'pic',           // Maps to a custom pic property of the item data
            properties: {            // Sets the image view  properties
                width: '50dp', height: '50dp', left: 0
            }
        },
        {                            // Title
            type: 'Ti.UI.Label',     // Use a label for the title
            bindId: 'info',          // Maps to a custom info property of the item data
            properties: {            // Sets the label properties
                color: 'black',
                font: { fontFamily:'Arial', fontSize: '20dp', fontWeight:'bold' },
                left: '60dp', top: 0,
            }
        },
        {                            // Subtitle
            type: 'Ti.UI.Label',     // Use a label for the subtitle
            bindId: 'es_info',       // Maps to a custom es_info property of the item data
            properties: {            // Sets the label properties
                color: 'gray',
                font: { fontFamily:'Arial', fontSize: '14dp' },
                left: '60dp', top: '25dp',
            }
        }
    ]
};

var listView = Ti.UI.createListView({
    templates: { 'template': myTemplate },
    defaultItemTemplate: 'template'
});
var sections = [];

var fruitSection = Ti.UI.createListSection({headerTitle: 'Fruits / Frutas'});
var fruitDataSet = [
    {
        info: {
            text: 'Apple',
            accessibilityLabel: 'AppleLabel,'
        },
        es_info: {
            text: 'Manzana',
            accessibilityHint: 'Manzanahint;'
        },
        pic: {image: 'apple.png',
            accessibilityValue: 'apple.pngVALUE.'}
    },
    {
        info: {text: 'Banana',
            accessibilityLabel: 'BananaLabel,',
            accessibilityValue: 'BananaVALUE.'},
        es_info: {text: 'Banana',
            accessibilityHint: 'Bananahint;',
            accessibilityValue: 'BananaVALUE.'},
        pic: {image: 'banana.png',
            accessibilityLabel: 'banana.pngLabel,',
            accessibilityHint: 'banana.pnghint;'}
    }
];
fruitSection.setItems(fruitDataSet);
sections.push(fruitSection);
listView.setSections(sections);
win.add(listView);
win.open();
```


![timob-24415](https://user-images.githubusercontent.com/7796592/58818599-f6e3f000-8636-11e9-864e-8d169aa95c3b.png)

garymathews commented:
Closing in favor of https://github.com/appcelerator/titanium_mobile/pull/10978
